app/views.py

```diff
@@ -1,55 +1,70 @@
 from app.serializers import UserSerializer
 from .models import User
 from rest_framework.views import APIView
-from . import serializers
 from rest_framework import status
 from rest_framework.response import Response
 from rest_framework.authtoken.models import Token
 from django.contrib.auth import authenticate
-from rest_framework import authentication
-# Create your views here.
+from rest_framework import authentication, permissions
+from django.shortcuts import get_object_or_404
+from .utils import admin_required, manager_required
+
 
 class Userview(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    permissions_classes = [permissions.IsAuthenticated, admin_required, manager_required]
     serializer_class = UserSerializer
     
     
+    
     def get(self, request, id=None):
-        #its for user id 
         if id:
             try:
-                user = User.objects.get(id=id)
+                user = get_object_or_404(User, id)
                 serializer = self.serializer_class(user)
                 return Response(serializer.data)
             except User.DoesNotExist:
                 return Response({'details': 'does not found'}, status=status.HTTP_404_NOT_FOUND)
-        #its for user list
         else:
-            user = User.objects.all()
+            user = User.objects.filter()
             serializer = self.serializer_class(user, many=True)
             return Response(serializer.data)
                     
     
     def post(self, request):
-        serializer = UserSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
-        return Response({"massage" : "Details Not found"}, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        user = User.objects.create_user(
+            name = request.data.get('name'),
+            email = request.data.get('email'),
+            password = request.data.get('password'),
+            role = request.data.get('role', 'User')
+        )
+        serializer =  UserSerializer(user)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+        #return Response(serializer.errors, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
             
     
-    def put(self, request, id=id):
-        user = User.objects.get(id=id)
-        serializer = UserSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
+    def put(self, request, id=None):
+        try:
+            user = User.objects.update(
+                name = request.data.get('name'),
+                email = request.data.get('email'),
+                password = request.data.get('password'),
+                role = request.data.get('role')
+            )
+            #user = User.objects.get(id)
```

Comment:
What's the purpose of serializer if you are updating it using model method?
Use serializer for it.